### ISSUE TITLE
FormArray & ControlValueAccessor does not pass through validation when sub component appears.

### ISSUE THREAD

**Issue by gerhardboer** [angular/angular#32454]
# 🐞 bug report

### Affected Package

The issue is caused by package @angular/forms

### Is this a regression?
No

 Yes, the previous version in which this bug was not present was: ....

### Description

 A clear and concise description of the problem...
In a complex form environment, i have a formarray inside a ControlValueAccessor (CVA), and another CVA with some internal validation.
The CVA with internal validation is shown optionally with an ngIf, and can be activated from the same FormGroup in the FormArray.

If i use the CVA in a FormArray that's directly connected to the main Form, the main form becomes invalid when the CVA appears, and valid when the CVA is removed.

However, when i use the same construction, but the FormArray is also in a CVA, the main form does not become invalid when the CVA appears.

in both situation the statusChanges event emitter is not activated, until the CVA with the validation is interacted with.

## 🔬 Minimal Reproduction

https://stackblitz.com/github/gerhardboer/formarray-cva-bug
the Broken module shows the issue
the Working module shows it working when no CVA boundary is present.

## 🔥 Exception or Error
the main form does not become invalid when a CVA with internal validation is added from a FormArray from a CVA

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular 8
</code></pre>

**Anything else relevant?**
This issue might be related: https://github.com/angular/angular/issues/14542

**Comment by AndrewKushnir** [comment]
Hi @gerhardboer, thanks for reporting the issue.

I checked the repro and it looks quite complex. Could you please try to simplify it and keep the minimum possible setup that is needed to reproduce the issue?

Thank you.

**Comment by gerhardboer** [comment]
Hi @AndrewKushnir 

I've tried to simplify and explain the repo a bit: https://stackblitz.com/edit/simplified-form-cva-bug

Unfortunately, it's a complex situation. I hope it's more clear now.
I've also tried updating to Angular 11, since there was an update to statusChanges, which might have fixed this problem. But it did not seem to work.

The point of the functionality is to add required ControlValueAccessors to a form, which should make the parent form invalid on appearance.
But when using a ControlValueAccessor in between the required field and the parent form, the parent form validity state is not updated correctly.

I hope this helps, it's been a while (exactly 1 year i noticed 🎂) since the issue was submitted and i'm not even currently at the same job where the issue originated from.
